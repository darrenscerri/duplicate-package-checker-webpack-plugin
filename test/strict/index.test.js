var webpack = require("webpack");
var assert = require("assert");
var MakeConfig = require("./make.webpack.config");

describe("Simple dependency tree", function() {
  it("should output warnings if strict", function(done) {
    webpack(MakeConfig(), function(err, stats) {
      assert(stats.compilation.warnings.length === 2);
      expect(stats.compilation.warnings[0].message).toMatchSnapshot();
      expect(stats.compilation.warnings[1].message).toMatchSnapshot();
      done();
    });
  });

  it("should not output warnings if not strict", function(done) {
    webpack(
      MakeConfig({
        strict: false
      }),
      function(err, stats) {
        assert(stats.compilation.warnings.length === 2);
        expect(stats.compilation.warnings[0].message).toMatchSnapshot();
        expect(stats.compilation.warnings[1].message).toMatchSnapshot();
        done();
      }
    );
  });
});
