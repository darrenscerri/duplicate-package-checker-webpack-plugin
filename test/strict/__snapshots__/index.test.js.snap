// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple dependency tree should not output warnings if not strict 1`] = `
"a
[0m  Multiple versions of [0m[32m[1ma[22m[39m[37m found:[39m
[37m[39m    [32m[1m1.0.0[22m[39m [37m[1m./~/a[22m[39m
    [32m[1m2.0.0[22m[39m [37m[1m./~/b/~/a[22m[39m
"
`;

exports[`Simple dependency tree should not output warnings if not strict 2`] = `
"b
[0m  Multiple versions of [0m[32m[1mb[22m[39m[37m found:[39m
[37m[39m    [32m[1m1.0.0[22m[39m [37m[1m./~/b[22m[39m
    [32m[1m2.0.0[22m[39m [37m[1m./~/c/~/a/~/b[22m[39m

[37m[1mCheck how you can resolve duplicate packages: [22m[39m
https://github.com/darrenscerri/duplicate-package-checker-webpack-plugin#resolving-duplicate-packages-in-your-bundle
"
`;

exports[`Simple dependency tree should output warnings if strict 1`] = `
"a
[0m  Multiple versions of [0m[32m[1ma[22m[39m[37m found:[39m
[37m[39m    [32m[1m1.0.0[22m[39m [37m[1m./~/a[22m[39m
    [32m[1m2.0.0[22m[39m [37m[1m./~/b/~/a[22m[39m
    [32m[1m2.5.0[22m[39m [37m[1m./~/c/~/a[22m[39m
"
`;

exports[`Simple dependency tree should output warnings if strict 2`] = `
"b
[0m  Multiple versions of [0m[32m[1mb[22m[39m[37m found:[39m
[37m[39m    [32m[1m1.0.0[22m[39m [37m[1m./~/b[22m[39m
    [32m[1m2.0.0[22m[39m [37m[1m./~/c/~/a/~/b[22m[39m

[37m[1mCheck how you can resolve duplicate packages: [22m[39m
https://github.com/darrenscerri/duplicate-package-checker-webpack-plugin#resolving-duplicate-packages-in-your-bundle
"
`;
